src/Cli/dotnet/commands/dotnet-workload/uninstall/WorkloadUninstallCommand.cs

```diff
@@ -43,47 +44,61 @@ internal class WorkloadUninstallCommand : WorkloadCommandBase
             };
 
             var creationResult = WorkloadResolverFactory.Create(creationParameters);
+            _workloadResolver = creationResult.WorkloadResolver;
 
             _sdkVersion = creationResult.SdkVersion;
 
             var sdkFeatureBand = new SdkFeatureBand(_sdkVersion);
-            _workloadInstaller = WorkloadInstallerFactory.GetWorkloadInstaller(Reporter, sdkFeatureBand, creationResult.WorkloadResolver, Verbosity, creationResult.UserProfileDir, VerifySignatures, PackageDownloader, creationResult.DotnetPath);
+            _workloadInstaller = WorkloadInstallerFactory.GetWorkloadInstaller(Reporter, sdkFeatureBand, _workloadResolver, Verbosity, creationResult.UserProfileDir, VerifySignatures, PackageDownloader, creationResult.DotnetPath);
         }
 
         public override int Execute()
         {
+            WorkloadHistoryRecorder recorder = new WorkloadHistoryRecorder(_workloadResolver, _workloadInstaller);
+            recorder.HistoryRecord.CommandName = "uninstall";
+            recorder.HistoryRecord.WorkloadArguments = _workloadIds.Select(id => id.ToString()).ToList();
```

Comment:
I noticed a pattern where in each `Execute` method for the workload commands, you're creating a new `WorkloadHistoryRecorder` and then using it. Instead, why not make a `WorkloadHistoryRecorder` a property of the base `WorkloadCommandBase` and then use that property when it is necessary in each command? You'd have to send the necessary information to the constructor of `WorkloadCommandBase` to make a new recorder.

Reply:
One of the arguments to make a WorkloadHistoryRecorder is the _workloadInstaller, and though that's present on WorkloadCommandBase, it's initialized in the constructor for each command separately, so I don't think I could move the WorkloadHistoryRecorder's initialization all the way to the base's constructor. I think I could move it back to the constructor, but I'm not sure I see a lot of added value in doing that.

Did I miss something?

Reply:
Doing a quick search in the PR, it looks like `InstallingWorkloadCommand` uses `WorkloadCommandBase` but doesn't create a `WorkloadHistoryRecorder`. That seems a bit... odd, right? I was looking for uses of `WorkloadCommandBase` in the PR for context.

For this pattern, since it only relies on `_workloadResolver` and `_workloadInstaller` to create a `WorkloadHistoryRecorder` and both of those are created in the constructor, it looks logical to make an class-wide instance of `WorkloadHistoryRecorder`, right? Do you need a new one each call to `Execute`? That's generally how to think about it. What is the lifetime necessary for it? Does it bind to the class lifetime or `Execute`'s lifetime?

Reply:
Aren't they the same anyway? When someone calls a command, say, `dotnet workload install <args>`, it initializes the class then calls execute. Once execute is finished, so is the command, and it disposes of everything, right? So I think they're equivalent performance-wise.

Reply:
> So I think they're equivalent performance-wise.

I wasn't particularly concerned about performance in this situation. It is more about code duplication and maintenance. If you can create something one place and reuse it in multiple places in the code, it is (generally) more maintainable. But I'm trying to rationalize if it makes sense here, as returning to this 3 weeks later is giving my brain a bit of stretching to remember. 😅

>Aren't they the same anyway?

So, I was thinking about it and these classes can *technically* be used for purposes other than running execute. But I don't have any examples of this. Even the `dotnet restore` is a completely separate command than the restore operation logic (called `RestoringCommand`) that is used by things like Build, Pack, Publish, etc.

I think the primary purpose of my original comment here wasn't on it causing any functional differences. I just saw a pattern that involved duplication and wondered if it was necessary to do that duplication. Base classes usually handle that burden of reducing duplication. Consolidating boilerplate helps maintainability, generally.